pkg/virt-controller/services/template_test.go

```diff
@@ -2229,6 +2229,58 @@ var _ = Describe("Template", func() {
 				Expect(err).ToNot(HaveOccurred())
 				Expect(pod.Spec.Containers[0].Resources.Requests.Cpu().String()).To(Equal("150m"))
 			})
+
+			It("should honor memoryOvercommit when set in the CR", func() {
+				config, kvStore, svc = configFactory(defaultArch)
+
+				By("Creating a VMI")
+				vmi := v1.VirtualMachineInstance{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testvmi",
+						Namespace: "default",
+						UID:       "1234",
+					},
+					Spec: v1.VirtualMachineInstanceSpec{
+						Domain: v1.DomainSpec{
+							Memory: &v1.Memory{
+								Guest: pointer.P(resource.MustParse("1Gi")),
+							},
+							Resources: v1.ResourceRequirements{
+								Requests: k8sv1.ResourceList{
+									// This would usually be set by the mutating webhook
+									k8sv1.ResourceMemory: resource.MustParse("1Gi"),
+								},
+							},
+						},
+					},
+				}
+
+				By("Checking how much memory the pod requests by default")
+				pod, err := svc.RenderLaunchManifest(&vmi)
+				Expect(err).ToNot(HaveOccurred())
+				mem100 := pod.Spec.Containers[0].Resources.Requests.Memory()
+
+				By("Setting a memory overcommit of 110% in the CR")
+				kvConfig := kv.DeepCopy()
+				kvConfig.Spec.Configuration.DeveloperConfiguration.MemoryOvercommit = 110
+				testutils.UpdateFakeKubeVirtClusterConfig(kvStore, kvConfig)
+
+				By("Checking how much memory the pod requests now")
```

Comment:
race condition possible? Should we double check the CR actually reflects 110 before trying to re-render the manifest?

Reply:
This is a unit test, and I believe `UpdateFakeKubeVirtClusterConfig` updates things instantly (dozens of other tests make the same assumption).